paasta_tools/setup_kubernetes_job.py

```diff
@@ -128,34 +161,60 @@ def validate_job_name(service_instance: str) -> bool:
     return True
 
 
+def get_kubernetes_deployment_config(
+    service_instances_with_valid_names: list,
+    cluster: str,
+    soa_dir: str = DEFAULT_SOA_DIR,
+) -> List[Tuple[bool, KubernetesDeploymentConfig]]:
+    service_instance_configs_list = []
+    for service_instance in service_instances_with_valid_names:
+        try:
+            service_instance_config = load_kubernetes_service_config_no_cache(
+                service=service_instance[0],
+                instance=service_instance[1],
+                cluster=cluster,
+                soa_dir=soa_dir,
+            )
+            service_instance_configs_list.append((True, service_instance_config))
+        except NoDeploymentsAvailable:
+            log.debug(
+                "No deployments found for %s.%s in cluster %s. Skipping."
+                % (service_instance[0], service_instance[1], cluster)
+            )
+            return [(True, None)]
+        except NoConfigurationForServiceError:
+            error_msg = (
+                f"Could not read kubernetes configuration file for %s.%s in cluster %s"
+                % (service_instance[0], service_instance[1], cluster)
+            )
+            log.error(error_msg)
+            return [(False, None)]
```

Comment:
I meant to return 1-element list. The reason I meant to do this is I thought if we should exit right away if we have one instance that have NoDeploymentsAvailable or NoConfigurationForServiceError. But now that I'm thinking about it we shouldnt ignore other instances that could be processed especially if skj runs every 30s

Reply:
Yeah I think it's probably better to process the instances that we can: the `NoDeploymentsAvailable` situation will happen every time that someone defines a new service in yelpsoa-configs until their deploy pipeline succeeds for the first time - this can take a long time, and we don't want to block deployments of all other services in the mean time.